src/scripts/main.js

```diff
@@ -1 +1,54 @@
 'use strict';
+
+const firstPromise = new Promise(function (resolve, reject) {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+
+    setTimeout(() => {
+      reject(new Error(`First promise was rejected`));
+    }, 3000);
+  });
+});
+
+const secondPromise = new Promise(function (resolve) {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  });
+});
+
+const thirdPromise = new Promise(function (resolve, reject) {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 3) {
+      resolve('Third promise was resolved');
+    }
+
+    reject(new Error('Third promise was resolved'));
```

Comment:
The third promise has a logical error. It resolves when the middle mouse button (button 3) is clicked, but it also immediately rejects with an error message stating 'Third promise was resolved'. This seems contradictory. You should decide whether this promise should resolve or reject based on the event.